Easydict/objc/Service/DeepL/EZDeepLTranslate.m

```diff
@@ -154,59 +136,29 @@ - (BOOL)autoConvertTraditionalChinese {
     return YES;
 }
 
-#pragma mark - WebView Translate
-
-- (void)webViewTranslate:(nonnull void (^)(EZQueryResult *, NSError *_Nullable))completion {
-    NSString *wordLink = [self wordLink:self.queryModel];
-    
-    mm_weakify(self);
-    [self.queryModel setStopBlock:^{
-        mm_strongify(self);
-        [self.webViewTranslator resetWebView];
-    } serviceType:self.serviceType];
-    
-    [self.webViewTranslator queryTranslateURL:wordLink completionHandler:^(NSArray<NSString *> *_Nonnull texts, NSError *_Nonnull error) {
-        if ([self.queryModel isServiceStopped:self.serviceType]) {
-            return;
-        }
-
-        self.result.translatedResults = texts;
-        completion(self.result, error);
-    }];
-
-    //    CFAbsoluteTime startTime = CFAbsoluteTimeGetCurrent();
-    //    NSString *monitorURL = @"https://www2.deepl.com/jsonrpc?method=LMT_handle_jobs";
-    //    [self.webViewTranslator monitorBaseURLString:monitorURL
-    //                                         loadURL:self.wordLink
-    //                               completionHandler:^(NSURLResponse *_Nonnull response, id _Nullable responseObject, NSError *_Nullable error) {
-    //        CFAbsoluteTime endTime = CFAbsoluteTimeGetCurrent();
-    //        MMLogInfo(@"API deepL cost: %.1f ms", (endTime - startTime) * 1000); // cost ~2s
-    //
-    //        MMLogInfo(@"deepL responseObject: %@", responseObject);
-    //    }];
-}
-
 #pragma mark - DeepL Web Translate
 
 /// DeepL web translate. Ref: https://github.com/akl7777777/bob-plugin-akl-deepl-free-translate/blob/9d194783b3eb8b3a82f21bcfbbaf29d6b28c2761/src/main.js
 - (void)deepLWebTranslate:(NSString *)text from:(EZLanguage)from to:(EZLanguage)to completion:(void (^)(EZQueryResult *, NSError *_Nullable))completion {
-    NSString *souceLangCode = [self languageCodeForLanguage:from];
+    NSString *sourceLangCode = [self languageCodeForLanguage:from];
+    sourceLangCode = [self removeLanguageVariant:sourceLangCode];
+
     NSString *regionalVariant = [self languageCodeForLanguage:to];
     NSString *targetLangCode = [regionalVariant componentsSeparatedByString:@"-"].firstObject; // pt-PT, pt-BR
 
     NSString *url = @"https://"
-                    @"www2."
-                    @"deepl.com"
-                    @"/jsonrpc";
-    
+    @"www2."
+    @"deepl.com"
+    @"/jsonrpc";
```

Comment:
I'm curious here why not make them a string like `@"https://www2.deepl.com/jsonrpc"` here, and I'm very surprised it's a valid declaration in objc.

Reply:
<img width="747" alt="Screenshot 2024-11-01 at 14 54 26" src="https://github.com/user-attachments/assets/7bf4fe75-f677-4a5f-b189-852efe0aa518">

This is called "string literal concatenation" in C/Objective-C. Nice to know this 👨🏽‍💻

Reply:
Because we are using the free web API from DeepL here, I am worried they might cause me trouble, so I am using this concatenation method to avoid being searched.

This is a trick for splitting strings  [how-to-split-a-string-literal-across-multiple-lines-in-c-objective-c](https://stackoverflow.com/questions/797318/how-to-split-a-string-literal-across-multiple-lines-in-c-objective-c/797351#797351)